client.go

```diff
@@ -119,6 +127,44 @@ func getImageFromSource(ctx context.Context, imgStr string, source image.Source,
 	return nil, fmt.Errorf("unable to detect input for '%s', errs: %w", imgStr, errors.Join(errs...))
 }
 
+func getImageSizeFromSource(ctx context.Context, imgStr string, source image.Source, options ...Option) (int64, error) {
+	log.Debugf("image: source=%+v location=%+v", source, imgStr)
+
+	// apply ImageProviderConfig config
+	cfg := config{}
+	if err := applyOptions(&cfg, options...); err != nil {
+		return 0, err
+	}
+
+	// select image provider
+	providers := collections.TaggedValueSet[image.Provider]{}.Join(
+		ImageProviders(ImageProviderConfig{
+			UserInput: imgStr,
+			Platform:  cfg.Platform,
+			Registry:  cfg.Registry,
+		})...,
+	)
+	if source != "" {
+		source = strings.ToLower(strings.TrimSpace(source))
+		providers = providers.Select(source)
+		if len(providers) == 0 {
+			return 0, fmt.Errorf("unable to find image providers matching: '%s'", source)
+		}
+	}
+
+	var errs []error
+	for _, provider := range providers.Values() {
+		compressedSize, err := provider.ImageSize(ctx)
```

Comment:
Instead of calling provider.provide which pulls the image, this will only get the data from the manifest